test/e2e/operator/operator.go

```diff
@@ -0,0 +1,293 @@
+				del(dep.ObjectMeta, func() error {
+					framework.Logf("Deleting deployment '%s/%s'", dep.Namespace, dep.Name)
+					return cs.AppsV1().Deployments(dep.Namespace).Delete(dep.Name, nil)
+				})
+			}
+
+			// Delete the namespace
+			if deletableNamespace(o.Namespace) {
+				if ns, err := cs.CoreV1().Namespaces().Get(o.Namespace, metav1.GetOptions{}); !getFailure(err) {
+					del(ns.ObjectMeta, func() error {
+						framework.Logf("Deleting namespace '%s'", ns.Name)
+						return cs.CoreV1().Namespaces().Delete(o.Namespace, nil)
+					})
+				}
+			}
+		}
+
+		if done && success {
+			framework.Logf("Deletion SUCCESS!!!")
+			return nil
+		}
+
+		// The actual API calls above are quick, actual deletion
+		// is slower. Here we wait for a short while and then
+		// check again whether all objects have been deleted.
+		select {
+		case <-ticker.C:
+		case <-deadline.Done():
+			return fmt.Errorf("timed out while trying to delete the PMEM-CSI operator in namespace %q", o.Namespace)
+		}
+	}
+}
+
+// FindOperatorDeployment checks whether there is a PMEM-CSI operator
+// installation in the cluster. An installation is found via its
+// deployment name and namespace.
+func FindOperatorDeployment(c *deploy.Cluster, o *Operator) (*appsv1.Deployment, error) {
+	cs := c.ClientSet()
+	framework.Logf("Checking if the operator '%s/%s' running", o.Namespace, o.Name)
+	dep, err := cs.AppsV1().Deployments(o.Namespace).Get(o.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return dep, nil
+}
+
+// EnsureOperatorDeployed registers a BeforeEach function which will ensure that when
+// a test runs, the desired deployment exists. Deployed drivers are intentionally
+// kept running to speed up the execution of multiple tests that all want the
+// same kind of deployment.
+func EnsureOperatorDeployed(c *deploy.Cluster, o *Operator) error {
+	dep, err := FindOperatorDeployment(c, o)
+	framework.ExpectNoError(err, "check for PMEM-CSI driver")
+	if dep != nil {
+		framework.Logf("delete existing operator deployment '%s/%s'", dep.Namespace, dep.Name)
+		// Currently all deployments share the same driver name.
+		err := EnsureOperatorRemoved(c, o)
+		framework.ExpectNoError(err, "remove PMEM-CSI deployment")
+	}
+
+	// At the moment, the only supported deployment method is via test/setup-deployment.sh.
+	cmd := exec.Command("test/start-operator.sh")
+	cmd.Dir = os.Getenv("REPO_ROOT")
+	cmd.Env = os.Environ()
+	cmd.Stdout = ginkgo.GinkgoWriter
+	cmd.Stderr = ginkgo.GinkgoWriter
+	err = cmd.Run()
+
+	framework.ExpectNoError(err, "create %s PMEM-CSI deployment", o.Name)
+
+	WaitForOperator(c, o)
+
+	return nil
+}
+
+var tests = map[string]func(o *Operator, f *framework.Framework){}
+
+// DescribeForAll registers tests like gomega.Describe does.
+func DescribeForAll(what string, f func(o *Operator, f *framework.Framework)) bool {
```

Comment:
This code looks confusing and unnecessarily complex when looking at this file because there is just one variant that each test gets registered for.

I guess the intention is to do something as in `deploy/deploy.go` and add more variants for deploying the operator, but perhaps add a comment here that explains the API and how it will evolve in the future so that others without that background knowledge also understand it?

Or do something entirely differently... see my other comment.